src/app/components/control-bar/control-bar.component.ts

```diff
@@ -376,319 +377,117 @@ export class ControlBarComponent implements OnInit {
-                group.icon +
-                '\r\n';
-              // And adds it to a growing full CSV file
-              CSVHolder += tempCSV;
-            }
-          });
-        }
+        this.download(blob, '.json', filename);
       }
-    });
-    let content;
-    let extension;
-    // determine whether the file is wanted as a JSON or a CSV
-    if (isJSON) {
-      content = JSONHolder;
-      extension = '.json';
     } else {
-      content = CSVHolder;
-      extension = '.csv';
-    }
+      const csvFiles = exportList.map((exportItem) => {
+        const csvRows = [];
+        const headers = this.getCSVHeaders(exportItem.features);
+        csvRows.push(headers.join(','));
+
+        exportItem.features.forEach((ef) => {
+          const values = headers.map((header) =>
+            ef[header] ? ef[header] : ''
+          );
+          csvRows.push(values.join(','));
+        });
 
-    // If the function is marked for export to Design Safe, route through export, otherwise, download the file
-    if (forExport) {
-      fileName == ''
-        ? (fileName = projID + extension)
-        : (fileName += extension);
-      this.filesService.export(systemID, path, fileName, extension, content);
-    } else {
-      this.download(content, extension, projID);
-    }
-  }
+        return csvRows.join('\n');
+      });
 
-  compileJSON(
-    coordinates,
-    featureID,
-    featureSource: string,
-    groups = [],
-    tags = [],
-    style
-  ) {
-    let compiledJSON = '';
-    let transferJSON;
-
-    // Add the most basic information to the compiled JSON
-    transferJSON = {
-      longitude: coordinates[0],
-      latitude: coordinates[1],
-      src: featureSource,
-    };
-    compiledJSON += JSON.stringify(transferJSON);
-
-    if (groups.length != 0) {
-      groups.forEach((group) => {
-        // At this point, group info should be added,
-        // If tags are set to a default value, there are none present, compile without tag information
-        transferJSON = {
-          groupName: group.name,
-          groupColor: group.color,
-        };
-        compiledJSON += JSON.stringify(transferJSON);
-        if (tags.length > 0) {
-          // Compile a JSON with full tag information
-          tags.forEach((tag) => {
-            if (tag.feature == featureID) {
-              transferJSON = {
-                icon: style.faIcon,
-                'icon color': style.color,
-                'tag name': tag.label,
-                'tag type': tag.type,
-                'tag selection': tag.extra[0].option,
-              };
-              compiledJSON += JSON.stringify(transferJSON);
-            }
-          });
+      const zip = new JSZip();
+      csvFiles.forEach((csv, i) => {
+        zip.file(`group-${i}.csv`, csv);
+      });
+
+      zip.generateAsync({ type: 'blob' }).then((content) => {
+        if (forExport) {
+          this.filesService.export(systemID, path, fileName, '.zip', content);
```

Comment:
Export to DesignSafe:
* file exported is not csv but maybe the zip. should the UI be updated
 to reflect that?
* "If no filename is specified, it will default to the Gallery ID number." I don't think this is no longer true
* when saving a file to DesignSafe if you don't change the name and leave it as "Custom File Name", then it fails as the formname has filename=""

Reply:
Adjusted so that it follows the scheme of CSV/JSON export as a fallback.

Reply:
I don't think any of these are addressed.